### PR TITLE
Add support for I2S MCLK.

### PR DESCRIPTION
**Pull request by DatanoiseTV**
This commit implements support for generating basic 256*fs MCLK using the APLL.

### CODE REVIEW

File: src/AudioOutputI2S.cpp [hunk 1]
```diff
@@ -104,6 +104,20 @@ bool AudioOutputI2S::SetPinout(int bclk, int wclk, int dout)
 
   return true;
 }
+
+bool AudioOutputI2S::SetPinout(int bclk, int wclk, int dout, int mclk)
```
Comment (earlephilhower) [inline src/AudioOutputI2S.cpp:108]: Can you please add `(void) mclk` in the non-ESP32 path?  OTW we get fatal warning/errors.

File: src/AudioOutputI2S.cpp [hunk 2]
```diff
@@ -147,6 +161,16 @@ bool AudioOutputI2S::SetLsbJustified(bool lsbJustified)
   return true;
 }
 
+bool AudioOutputI2S::SetMclk(bool enabled){
```
Comment (earlephilhower) [inline src/AudioOutputI2S.cpp:164]: Same as other comment, for `enabled`.

**Review by earlephilhower — CHANGES_REQUESTED** [review]

File: src/AudioOutputI2S.cpp (hunk 2 above)
Comment (DatanoiseTV) [reply src/AudioOutputI2S.cpp:166]: Done.

**Comment by earlephilhower** [comment]
Something got weird on the commit/pushes and your `(void)` marker seems to have been lost (see the 8266 failures).  Can you take a look again, please?

**Comment by DatanoiseTV** [comment]
> Something got weird on the commit/pushes and your `(void)` marker seems to have been lost (see the 8266 failures). Can you take a look again, please?

I think now it is right.

**Comment by earlephilhower** [comment]
OK, we're almost there.  The change in the 8266 needs to applied to the RP2040 branch and we should all be OK.  The joys of supporting 3 different (actually w/the ESP32-XXs maybe 6?) boards.

**Comment by earlephilhower** [comment]
@DatanoiseTV can you pull in the latest `master`?    The ESP8266 internal test scripts changed over in that repo, so I needed to hack the tests here that used it.  It should now work and we should be able to hopefully get a complete run.

**Review by earlephilhower — APPROVED** [review]

**Comment by FREEWING-JP** [comment]
I tested CS4344 I2S DAC with ESP32-D0WD-V3 (revision v3.0) .
* CS4344 need MCLK
It works Fine !
Thank you !

**Pull request merged**